pentaho-cache-manager/impl/ehcache/src/main/java/org/pentaho/caching/ehcache/EhcacheProvidingService.java

```diff
@@ -12,69 +12,25 @@
 
 package org.pentaho.caching.ehcache;
 
-import net.sf.ehcache.CacheManager;
-import net.sf.ehcache.config.Configuration;
-import org.ehcache.jcache.JCacheCachingProvider;
-import org.ehcache.jcache.JCacheManager;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.spi.CachingProvider;
 import org.pentaho.caching.api.PentahoCacheSystemConfiguration;
 import org.pentaho.caching.spi.AbstractCacheProvidingService;
 
-import java.net.URI;
-import java.util.Properties;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ThreadFactory;
-
 /**
  *  PentahoCacheProvidingService implementation which leverages the
  *  org.ehcache.jcache project to provide a JCache CacheManager wrapping
  *  ehcache 2.x. Once we move to ehcache 3.x we'll no longer need the wrapper.
  */
 public class EhcacheProvidingService extends AbstractCacheProvidingService {
 
-  private static final JCacheCachingProvider providerInstance = new JCacheCachingProvider();
   private static CacheManager cacheManager;
 
-  private final ExecutorService executorService = Executors.newSingleThreadExecutor( new ThreadFactory() {
-    private final ThreadFactory threadFactory = Executors.defaultThreadFactory();
-
-    @Override public Thread newThread( Runnable r ) {
-      Thread thread = threadFactory.newThread( r );
-      thread.setName( "ehcache-jcache" );
-      thread.setDaemon( true );
-      return thread;
-    }
-  } );
-
-  public static CacheManager getCacheManager() {
-    if ( cacheManager == null ) {
-      cacheManager = CacheManager.newInstance( new Configuration() );
-    }
-    return cacheManager;
-  }
 
   @Override public javax.cache.CacheManager createCacheManager( PentahoCacheSystemConfiguration systemConfiguration ) {
-
-    return new JCacheManager(
-        providerInstance,
-        getCacheManager(),
-        URI.create( getClass().getName() ),
-      new Properties() ) {
-      @Override public ExecutorService getExecutorService() {
-        return executorService;
-      }
-    };
-  }
-
-  public void shutdown() {
-    try {
-      getCacheManager().shutdown();
-    } catch ( Exception ignored ) {
-      // best effort
-    }
+    CachingProvider cachingProvider= Caching.getCachingProvider( this.getClass().getClassLoader() );
+    return cachingProvider.getCacheManager();
```

Comment:
Why not making this a single line?
`return Caching.getCachingProvider( this.getClass().getClassLoader() ).getCacheManager();`